docker/Dockerfile.root6

```diff
@@ -93,6 +93,7 @@ ENV LIBXML2_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/libxml2-2.9.10-
 ENV Vc_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/vc-1.4.1-xdb5jhivtn2rejvvfs2whrf4nx4lxwbi
 ENV FASTJET_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/fastjet-3.3.3-xj22tput5pd5atejy4igsrtgdtygi74u
 ENV GSL_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/gsl-2.6-ao26bt7pm6xxywo2q7dcmhhg2lp2inx7
+ENV VC_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/vc-1.4.1-xdb5jhivtn2rejvvfs2whrf4nx4lxwbi
```

Comment:
`Vc_DIR` is defined three lines above this line.

Reply:
I did not notice you defined a Vc_DIR in dockerfile.root6, but it is not defined in dockerfile.root6.
I defined VC_DIR in both files and use all capital letters to maintain to other variables you defined.

Reply:
I guess the main branch is still using the Vc 0.7.4 from ROOT5 and you redefined the Vc_DIR in dockerfile.root6 to just pick up Vc 0.7.4 in spack. Removing Vc 0.7.4 from ROOT5 and just using spack + VC_DIR would give us a unified usages of Vc.

Reply:
`Vc_DIR` is defined by the spack package:

https://github.com/star-bnl/star-spack/blob/fba3a3ffbafc0c77fa46085d807b717ca807f345/packages/vc/package.py#L31-L34

That is the reason why it does not have to be set explicitly in Dockerfile.root5
The ENV definitons you are proposing to change are not relevant for the build stage because these values get overwritten by loading the environment modules before the actual build command. Also in case of ROOT6 the default value defined by Vc module is currently being overwritten manually:

https://github.com/star-bnl/star-sw/blob/948846a8dc8b9a891f42397889c2dbdfa1c644ec/docker/Dockerfile.root6#L124-L126

What you actually want to do in the context of this StiCA upgrade is to remove the line manually overwriting Vc_DIR in Dockerfile.root6 and instead add a similar one in Dockerfile.root5.

If you really want to rename the environment variable from `Vc_DIR` to `VC_DIR` then you better start with the definition in the spack package. I don't care about such nitpicking changes and would accept them. Although, this would require a rebuild of the base image for our CI tests.

Reply:
I do not know "Vc_DIR" is define by spack and export to the CI environment. Is this also true for other libraries, like  "GSL_DIR"? It would be better if such information is available more explicitly. I do not have string preferences between "Vc_DIR" and "VC_DIR", but I do prefer consistency. I assume redefine Vc_DIR exlicityly in Dokerfile.root5 would not be problem, right?

Reply:
> Is this also true for other libraries, like "GSL_DIR"?

It is up to the package to set such variables. You'd need to check the Spack package file to know for sure.

> I assume redefine Vc_DIR exlicityly in Dokerfile.root5 would not be problem, right?

Again, it depends on where you set it... It might be slightly better to avoid setting anything by hand if we don't have to. For example, if the hash changes then you need to remember to update it, etc.

Reply:
In this case,  I prefer to define every XX_DIR in both Dockerfiles explicitly. To check and remember every such variables in spack is not attractive to me.

Reply:
I do know how does the dockerfile interactive with spack, but redefine 

ENV Vc_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/vc_-1.4.1-waaa6u2x4ofz4sv5hkrt5576vuioxl4q

in dockerfile.root5 does not work. It still points to 0.7.4.

Reply:
I have tried what you suggested
```
What you actually want to do in the context of this StiCA upgrade is to remove the line manually overwriting Vc_DIR in Dockerfile.root6 and instead add a similar one in Dockerfile.root5.
```
But root5 CI jobs still failed. Can you take a look? @plexoos

Reply:
Try patching it like this

```diff
diff --git a/docker/Dockerfile.root5 b/docker/Dockerfile.root5
index 1d3735f570..99b5136a03 100644
--- a/docker/Dockerfile.root5
+++ b/docker/Dockerfile.root5
@@ -121,6 +121,8 @@ diff --git a/asps/rexe/Conscript b/asps/rexe/Conscript \n\
  && patch -p1 < root4star_cons.patch

 RUN source /etc/profile \
+ # Override default Vc version
+ && export Vc_DIR=/opt/software/linux-scientific7-x86_64/gcc-4.8.5/vc_-1.4.1-waaa6u2x4ofz4sv5hkrt5576vuioxl4q/ \
  && cd /star-sw \
  && cons +StarVMC/Geometry \
  && cons %OnlTools \

```